sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SharedThriftServer.scala

```diff
@@ -69,23 +89,42 @@ trait SharedThriftServer extends SharedSparkSession {
   }
 
   private def startThriftServer(attempt: Int): Unit = {
-    logInfo(s"Trying to start HiveThriftServer2:, attempt=$attempt")
+    logInfo(s"Trying to start HiveThriftServer2: mode=$mode, attempt=$attempt")
     val sqlContext = spark.newSession().sqlContext
-    // Set the HIVE_SERVER2_THRIFT_PORT to 0, so it could randomly pick any free port to use.
+    sqlContext.setConf(ConfVars.METASTORECONNECTURLKEY.varname,
+      s"jdbc:derby:;databaseName=$metastorePath;create=true")
+    sqlContext.setConf(ConfVars.METASTOREURIS.varname, "")
```

Comment:
What's this for?

Reply:
Some test failures showed that the metastore client was trying to connect through `METASTOREURIS` which seems to be set by other tests. But now I guess this won't be necessary as we skip that part